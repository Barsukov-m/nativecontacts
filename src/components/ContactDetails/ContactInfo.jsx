import { Text, TouchableNativeFeedback, View } from 'react-native';
import styles from './styles';
import Icon from 'react-native-vector-icons/MaterialIcons';

const ContactInfo = ({ title, data, icon, actionButton }) => {
  return (
    <TouchableNativeFeedback>
      <View style={styles.contactInfoContainer}>
        <View style={styles.contactInfo}>
          <Icon name={icon} size={22} />
          <View>
            <Text style={styles.data}>{title}</Text>
            <Text style={styles.title}>{data}</Text>
          </View>
        </View>
        {actionButton}
      </View>
    </TouchableNativeFeedback>
  );
};

export default ContactInfo;
