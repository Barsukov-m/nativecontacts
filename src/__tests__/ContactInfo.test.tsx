import { render } from '@testing-library/react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import ContactInfo from '../components/ContactDetails/ContactInfo';

jest.mock('react-native-vector-icons/MaterialIcons');

describe('<ContactInfo />', () => {
	it('renders correctly', () => {
		const actionButton = <Icon name="message" size={22} />;
		const tree = render(
			<ContactInfo
				title="Phone"
				data="0386-6098000"
				icon="phone"
				actionButton={actionButton}
			/>
		);
		const { getByText } = tree;

		// Check for the contact info title
		expect(getByText('Phone')).toBeDefined();

		// Check for the data
		expect(getByText('0386-6098000')).toBeDefined();

		// Check for the action button
		expect(Icon).toHaveBeenCalledWith(
			expect.objectContaining({
				name: 'phone',
			}),
			{}
		);

		// Check for snapshot match
		expect(tree).toMatchSnapshot();
	});
});
