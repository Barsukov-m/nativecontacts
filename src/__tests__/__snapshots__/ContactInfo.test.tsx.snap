// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactInfo /> renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 15,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 20,
      }
    }
  >
    <Text>
      Icon
    </Text>
    <View>
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 16,
          }
        }
      >
        Phone
      </Text>
      <Text
        style={
          {
            "fontSize": 12,
            "marginTop": 2,
          }
        }
      >
        0386-6098000
      </Text>
    </View>
  </View>
  <Text>
    Icon
  </Text>
</View>
`;
